### DISCUSSION TITLE
Connnect to multiple databases using multiple knex instances ?

### DISCUSSION THREAD

**Discussion by VaZark** [directus/directus#12699] [Archive]
Using plain knex.js allows us to access multiple databases using separate knex instances and adding them to a connection pool. 

Will it be possible to enforce this behaviour by overriding the connection during startup using extensions? Or will it require changes in the backend core to take in array of DB configs ?

## Usecase / Ideal setup

- SaaS application with hybrid multitenancy where the same database schema is expected everywhere.
- Use common directus instances but have separate directus settings in each tenant

**Comment by rijkvanzanten** [comment]
Doesn't knex's pooling allow you to create a pool of connections to the same database, rather than having multiple databases connected at the same time? I don't recall ever seeing a way in knex's query building to select the database first, and then continue the query 🤔 

To answer the question: I would not recommend doing that. There might be a lot of other things in the system that currently assume only a single database is connected, including caches. Also, the database connection is kept alive between requests, so you might run into unexpected issues where the wrong database is used for a given request 🤔

**Reply by VaZark** [reply]
Assuming i maintain separate connection pools for each DB, and cache keys are prefixed like « DB_NAME_KEY », that should resolve the cache issue right ? Or are there a lot more moving parts that implicitly call the active connection ?

**Reply by rijkvanzanten** [reply]
I don't know! This is not something that's officially supported, and my gut feeling says you might run into a couple other unexpected issues as Directus wasn't designed to handle this use case out of the box 👍🏻
